### PR TITLE
Preparatory Improvements

### PR DESCRIPTION
**Pull request by hgenru**
Changelog:
* Replace i18n with typesafe-i18n
* Add logger with colors
* Move register script to script folder

### CODE REVIEW

File: src/i18n/ru/index.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import type { Translation } from '../i18n-types'
+
+const ru: Translation = {
+  commands: {
+    ping: {
+      description: 'Бот отвечает `pong!`',
+    },
+    help: {
+      description: '',
```
Comment (ReFruity) [inline src/i18n/ru/index.ts:9]: Add non-empty description

**Review by ReFruity — CHANGES_REQUESTED** [review]

File: src/i18n/ru/index.ts (hunk 1 above)
Comment (hgenru) [reply src/i18n/ru/index.ts]: Added

File: config.template.json [hunk 2]
```diff
@@ -7,6 +7,5 @@
   "dbName": "ezbot",
   "languageCode": "en-US",
   "locale": "en",
-  "botTimeout": 40000,
-  "shouldRegisterSlashCommandsOnStart": true
+  "botTimeout": 40000
```
Comment (ReFruity) [inline config.template.json:10]: Add log level

File: scripts/register-commands.ts [hunk 3]
```diff
@@ -17,3 +17,6 @@ export async function registerSlashCommands(): Promise<void> {
   // eslint-disable-next-line no-console
   console.log('Successfully reloaded application (/) commands.')
```
Comment (ReFruity) [inline scripts/register-commands.ts:18]: Use logger

**Review by ReFruity — CHANGES_REQUESTED** [review]

File: src/logger.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      // eslint-disable-next-line no-console
+      ? (message, props = '') => console.info(formatMessage('info', message), props)
+      : noopLogger,
+    warn: logLevel <= LogLevelCode.WARN
+      // eslint-disable-next-line no-console
+      ? (message, props = '') => console.warn(formatMessage('warn', message), props)
+      : noopLogger,
+    error: logLevel <= LogLevelCode.ERROR
+      // eslint-disable-next-line no-console
+      ? (message, props = '') => console.error(formatMessage('errr', message), props)
```
Comment (ReFruity) [inline src/logger.ts:41]: errr -> err

File: src/logger.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+const noopLogger = () => undefined
+export default function logger(loggerName: string): Logger {
+  const logLevel = LogLevelCode[config.logLevel]
+  const formatMessage = (level: string, message: string) => `[${level}:${loggerName}] ${message}`
+  return {
+    debug: logLevel <= LogLevelCode.DEBUG
+      // eslint-disable-next-line no-console
+      ? (message, props = '') => console.debug(formatMessage('debg', message), props)
```
Comment (ReFruity) [inline src/logger.ts:29]: debg -> dbg

File: src/logger.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      : noopLogger,
+    info: logLevel <= LogLevelCode.INFO
+      // eslint-disable-next-line no-console
+      ? (message, props = '') => console.info(formatMessage('info', message), props)
```
Comment (ReFruity) [inline src/logger.ts:33]: info -> inf

File: src/logger.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      : noopLogger,
+    warn: logLevel <= LogLevelCode.WARN
+      // eslint-disable-next-line no-console
+      ? (message, props = '') => console.warn(formatMessage('warn', message), props)
```
Comment (ReFruity) [inline src/logger.ts:37]: warn -> wrn

File: src/logger.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      : noopLogger,
+    error: logLevel <= LogLevelCode.ERROR
+      // eslint-disable-next-line no-console
+      ? (message, props = '') => console.error(formatMessage('errr', message), props)
+      : noopLogger,
+    fatal: logLevel <= LogLevelCode.FATAL
+      // eslint-disable-next-line no-console
+      ? (message, props = '') => console.error(formatMessage('fatl', message), props)
```
Comment (ReFruity) [inline src/logger.ts:45]: fatl -> ftl

**Review by ReFruity — CHANGES_REQUESTED** [review]

File: src/discord/listeners.ts [hunk 9]
```diff
@@ -50,8 +51,8 @@ async function onPresenceUpdate(oldPresence: Presence, newPresence: Presence): P
 async function onInteractionCreate(interaction: Discord.Interaction): Promise<void> {
   if (!interaction.isCommand()) return
 
-  const command = commandStore.get(interaction.commandName)
-
+  // interaction.commandName isn't allowed string literal, just use any to avoid error
+  const command = commandStore.get(interaction.commandName as any)
```
Comment (ReFruity) [inline src/discord/listeners.ts:55]: as AllCommandNames

File: src/i18n/en/index.ts [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import type { BaseTranslation } from '../i18n-types'
+
+const en: BaseTranslation = {
+  commands: {
+    ping: {
+      description: 'Bot responds with `pong`!',
```
Comment (ReFruity) [inline src/i18n/en/index.ts:6]:
```suggestion
      description: 'Bot responds with "Pong!"',
```

File: src/i18n/en/index.ts [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+    },
+    help: {
+      description: 'Displays help on bot commands',
```
Comment (ReFruity) [inline src/i18n/en/index.ts:9]:
```suggestion
      description: 'Displays help on bot commands.',
```

File: src/discord/listeners.ts [hunk 12]
```diff
@@ -50,8 +51,8 @@ async function onPresenceUpdate(oldPresence: Presence, newPresence: Presence): P
 async function onInteractionCreate(interaction: Discord.Interaction): Promise<void> {
   if (!interaction.isCommand()) return
 
-  const command = commandStore.get(interaction.commandName)
-
+  // interaction.commandName isn't allowed string literal, just use any to avoid error
```
Comment (ReFruity) [inline src/discord/listeners.ts:54]:
```suggestion
```

File: src/i18n/ru/index.ts [hunk 13]
```diff
@@ -0,0 +1,14 @@
+import type { Translation } from '../i18n-types'
+
+const ru: Translation = {
+  commands: {
+    ping: {
+      description: 'Бот отвечает `pong!`',
```
Comment (ReFruity) [inline src/i18n/ru/index.ts:6]:
```suggestion
      description: 'Бот отвечает "Pong!"',
```

File: src/i18n/ru/index.ts [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+    },
+    help: {
+      description: 'Отображает помощь по командам',
```
Comment (ReFruity) [inline src/i18n/ru/index.ts:9]:
```suggestion
      description: 'Отображает помощь по командам.',
```

**Review by ReFruity — CHANGES_REQUESTED** [review]

**Pull request merged**